response_operations_ui/views/accounts.py

```diff
@@ -0,0 +1,204 @@
+import logging
+from structlog import wrap_logger
+from flask import Blueprint, request, render_template, redirect, url_for, abort, flash, current_app as app
+from itsdangerous import URLSafeSerializer, BadSignature, BadData, SignatureExpired
+
+from response_operations_ui.forms import RequestAccountForm, CreateAccountForm
+from response_operations_ui.controllers import uaa_controller
+from response_operations_ui.controllers.notify_controller import NotifyController
+from response_operations_ui.common import token_decoder
+from response_operations_ui.exceptions.exceptions import NotifyError
+
+logger = wrap_logger(logging.getLogger(__name__))
+
+account_bp = Blueprint('account_bp', __name__, static_folder='static', template_folder='templates')
+
+
+@account_bp.route('/request-new-account', methods=['GET'])
+def get_request_new_account(form_errors=None, email=None):
+    form = RequestAccountForm(request.form)
+    template_data = {
+        "error": {
+            "type": form_errors
+        }
+    }
+    return render_template('request-new-account.html', form=form, data=template_data, email=email)
+
+
+@account_bp.route('/request-new-account', methods=['POST'])
+def post_request_new_account():
+    form = RequestAccountForm(request.form)
+    form.email_address.data = form.email_address.data.strip()
+    email = form.data.get('email_address')
+
+    if form.validate():
+        local_part, domain_part = email.rsplit('@', 1)
+        logger.info('This is the domain part', domain_part=domain_part)
+        if domain_part != 'ons.gov.uk' and domain_part != 'ext.ons.gov.uk' and domain_part != 'ons.fake':
+            logger.info('Account requested for non-ONS email address')
+            errors = {'email_address': ['Not a valid ONS email address']}
+            return get_request_new_account(form_errors=errors, email=email)
+    
+        password = form.data.get('password')
+        admin_password = app.config['CREATE_ACCOUNT_ADMIN_PASSWORD']
+        if password != admin_password:
+            logger.info('Invalid password provided for create account')
+            errors = {'password': ['Incorrect admin secret']}
+            return get_request_new_account(form_errors=errors, email=email)
```

Comment:
I've seen a bunch of this and I don't think it's the best way to do it.  I don't think (though if you can show me counter examples then I'd happily have my mind changed) that having this code call a function which is actually the code for the endpoint is good practice.  
The function signature looks weird and I think we'll have other problems in the future if a change was made to an endpoint that's being called this way, now we have to trust that the functionality isn't being broken anywhere else.

With the number of times you're calling this, it'd be better to have a `_handle_errors` function (or something less terribly named ^^) where you do basically the same thing that you do in get_request_new_account but have it not be a function that correlates to an endpoint.  That function can return a template and you'd have the same functionality but less risk for future development